### ISSUE TITLE
Register all CTT tags

### ISSUE THREAD

**Issue by denislepage** [MotusDev/Motus-TO-DO#517]
Before we start processing data more widely, we should ensure that we have as complete a list as possible of the CTT tags that we are likely to see. My idea of registering new unknown tags (referenced in #508 is a bit of a bust. It's worse with older CTT files, but we'll quickly end up with millions of invalid tags and inflated numeric tag ID's.

We should check with CTT if they can send us a list of all candidate life tags that have been produced so far, whether or not they are likely to be part of Motus.

**Comment by denislepage** [comment]
We talked to Bob today. He is going to put together a complete list of tags and send us a CSV file. We're also setting up a new table to track the master set of tags that we may be listening to. Those won't be added to the tags table until: 1) the tag registration API provides the tags details (including a motus project ID), or 2) we pick up a detection for that tag. In the latter case, a new tagID will be created, but without any other details, and will be assigned to project 0.

Any detection of tags that are not in the master list will be discarded/ignored, so keeping the list up-to-date will be important.

Paul will be setting up an API entry point that will allow CTT to keep the full list current for that purpose (e.g. as soon as those new numbers are assigned on their end, ideally). 

We will also need to modify a few things on the other registration proc (is it prAPIImportTagsCTT?)

- it should also call that new proc (prInsertTagsCttMaster) as a safeguard in case the tag wasn't already added before, and only if the tag manufacturer is CTT.

- the proc should be modified to update existing records rather than return a "tag-exists" error. For instance, if a tag ID was first created from a detection, we'll need to add new details when it gets fully registered by CTT.

- as mentioned in a separate issue, the tag can also be added to multiple projects. I have provisionally created a new table called tags_projects, but we'll have to investigate the implications. For now, the proc should make sure it adds all projects to that table. We can keep project_id in tags for the time being (the original project_id, I guess).

**Comment by StuMackenzie** [comment]
presumed resolved.
